Login

I-Sheng opened the pull request:
# Add Login feature & views

<!--- There is a login page and some functions -->

## Description

<!--- Describe your changes in detail -->

Add login.

## Motivation or Context

<!--- Login is require-->
<!--- If it fixes an open issue, please link to the issue here. -->

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

| Item          | Is checked? | Description                                        |
| ------------- | ----------- | -------------------------------------------------- |
| Explanation   | V          | My changes are well-explained.                     |
| Documentation |V           | My works are well-commented or clear.              |
| Tests         | V           | I have added tests or reports to cover my changes. |

I-Sheng committed:
naive login page

I-Sheng committed:
basic styling

I-Sheng committed:
delete the view that is not needed

I-Sheng committed:
fix size

I-Sheng committed:
combined auth backend program

I-Sheng committed:
fix conflict

I-Sheng committed:
fix router

I-Sheng committed:
fix some conflicts

I-Sheng committed:
fix conflicts

I-Sheng committed:
fix;

I-Sheng committed:
change path to /function

I-Sheng committed:
fix path

I-Sheng committed:
delete page.vue

I-Sheng committed:
fin

I-Sheng committed:
fix router

I-Sheng committed:
fix styling

AsherJingkongChen committed:
(adjust): Header.vue: Correct conditional display (temp)
- Wrapping cookies as store may fix better

(adjust): Router: Change name

(remove): Unused assets

AsherJingkongChen commented on src/assets/down.svg:
## Note

Unused assets should be removed

AsherJingkongChen commented on src/components/CustomInput.vue:
## Comment


Quite simple but password should be hidden

AsherJingkongChen commented on src/components/Header.vue:
## Notice

Login/Logout conditional display

AsherJingkongChen commented on src/functions/cookies.ts:
## Notice

Semantic naming

AsherJingkongChen replied:
## Notice

Plus, it should be a store

AsherJingkongChen commented on src/router/index.ts:
## Notice

Add name

AsherJingkongChen commented on src/views/Login.vue:
## Notice

Refresh page is required. `Router.push` does not refresh so user does not know the session status in the first load.

AsherJingkongChen committed:
Merge branch 'main-login' into login-origin

AsherJingkongChen commented:
## Status

Changes are required before merge, but I will work on them later. Plus, I need to integrate the auth/cookie feature into other places.

AsherJingkongChen committed:
(patch-remove)

AsherJingkongChen committed:
(patch-dep)

AsherJingkongChen committed:
(feat): cookies => authSession.ts: Make local session as store
- Affect header and login view

AsherJingkongChen committed:
(rename): Login => Auth entrance view

AsherJingkongChen committed:
(feat): Integrate auth with feature views

AsherJingkongChen committed:
(adjust): authSession.ts: Ensure user can logout

AsherJingkongChen committed:
(adjust): LatestGameView.vue: Uses Logo as the fallback image for arena team
- logo.png path was changed

(adjust): Semantic naming

AsherJingkongChen committed:
(feat): AuthEntranceView.vue: Make styling being **UI-Design-Compliant**
- NOTE: **This IS essential for the frontend team!**

(adjust): ViewTitle.vue: Wrap title (h1) to ensure style being consistent across views
- Uses $route.name as title value

AsherJingkongChen committed:
(remove): CustomInput.vue: No need to use <input> component

AsherJingkongChen commented on package.json:
## Good Patch

It is necessary to add runtime dependency

AsherJingkongChen commented on public/logo.png:
## Good Change

Since Logo may not change frequently.

AsherJingkongChen commented on src/components/Header.vue:
## Good Patch

Using universal alert or popup widget.

AsherJingkongChen commented on src/components/Header.vue:
## Good Patch

Using reactive state (Pinia store) instead of pure functions to be Vue-compliant

AsherJingkongChen commented on src/components/ViewTitle.vue:
## Good Adjustment

Ensuring the style being consistent across different views.

AsherJingkongChen commented on src/functions/cookies.ts:
## Note

It was renamed as `src/stores/authSession.ts`

AsherJingkongChen commented on src/router/index.ts:
## Good Naming

Using Noun instead of Verb to make the view name clearer.

AsherJingkongChen commented on src/stores/authSession.ts:
## Note

This is adapted from `src/functions/cookies.ts`.

AsherJingkongChen commented on src/views/AuthEntranceView.vue:
## Note

This is adapted from `src/views/Login.vue`

AsherJingkongChen commented on src/views/GameView.vue:
## Good Patch

Wrapping stateful functions as store is more convenient to use in Vue.

AsherJingkongChen commented on src/views/LatestGameView.vue:
## Good Feature

Alternative Image Source

AsherJingkongChen commented on src/views/Login.vue:
## Note

This was renamed to `src/views/AuthEntranceView.vue`
